test/Microsoft.AspNetCore.Mvc.Core.Test/ApplicationParts/AssemblyPartTest.cs

```diff
@@ -22,6 +22,20 @@ public void AssemblyPart_Name_ReturnsAssemblyName()
         }
 
         [Fact]
+        public void AssemblyPart_Types_ReturnsDefinedTypes()
+        {
+            // Arrange
+            var assembly = typeof(AssemblyPartTest).GetTypeInfo().Assembly;
+            var part = new AssemblyPart(assembly);
+
+            // Act
+            var types = part.Types;
+
+            // Assert
+            Assert.Equal(assembly.DefinedTypes, types);
```

Comment:
You have to look at the intellisense to be sure what xUnit does. It really **wants** to compare these as collections, but it's not always clear because there's `Assert.Equal(T t1, T t2)` and `Assert.Equal(IEnumerable<T> ts1, IEnumerable<T> ts2)`. You can always write `Assert.Equal<TypeInfo>(assembly.DefinedTypes, types);` to prove that it's doing what you want, but VS will want to 'simplify' this.